serial_linux.go

```diff
@@ -137,6 +137,17 @@ func (p *Port) Read(b []byte) (n int, err error) {
 	return p.f.Read(b)
 }
 
+func (p *Port) Read2(b []byte, idx, l int) (n int, err error) {
+	t := make([]byte, l)
+	rcnt, err := p.f.Read(t)
+
+	for i := 0; i < rcnt; i++ {
+		b[i+idx] = t[i]
```

Comment:
this should be the built-in copy()

Reply:
You are right, I think may be do this change: ```copy(buf[idx:], tbuf[:rcnt])```